src/main/java/com/zombiecastlerush/building/Combat.java

```diff
@@ -0,0 +1,62 @@
+package com.zombiecastlerush.building;
+
+
+import com.zombiecastlerush.entity.Role;
+import com.zombiecastlerush.util.Parser;
+import com.zombiecastlerush.util.Prompter;
+
+import java.util.List;
+import java.util.Random;
+
+public class Combat extends Challenge {
+    public Combat(String description) {
+        super(description);
+    }
+
+    public static void combat(Role player, Role enemy) {
+        String msg = "Welcome to Combat mode, what would you like to do?";
+        String combatChoice = Prompter.getUserInput(msg);
+        while (player.getHealth() > 0 && enemy.getHealth() > 0) {
+            System.out.println("fight" +
+                    "run-away");
+            if (combatChoice.isEmpty()) {
+                continue;
+            }
+            combatChoiceParser(combatChoice, player, enemy);
+        }
+    }
+
+    private static void combatChoiceParser(String choice, Role player, Role enemy) {
+        List<String> combatCommandList = Parser.parse(choice);
+        if (combatCommandList == null) {
+            System.out.println("That's not a valid command. Please try again ....");
+        } else if (combatCommandList.get(0).contains("fight")) {
+            playerFight(player, enemy);
+        }
+    }
+
+    static void playerFight(Role player, Role enemy) {
+        int playerHitPoints = new Random().nextInt(50) + 1;
```

Comment:
Why do you need to have playerHitPoints variable with random numbers, doesn't Player already have a health property?

Reply:
After reading the rest of the PR, I feel like overall the code makes more sense if you change some of the variable names.